### PR TITLE
mgr/dashboard: ceph dashboard i18ntool

### PR DESCRIPTION
**Pull request by Exotelis**
Adding the ceph dashboard i18n that manages the translation files of the ceph dashboard.
In version 1.0.0 it's just capable of merging the target translation files with the source file to add missing target tags.

Upcoming versions will also handle the file management via transifex.

### CODE REVIEW

**Comment by LenzGr** [comment]
Thanks for submitting this proposal. On which branch do I have to perform this task? What happens if I want to maintain different versions of the translations, e.g. one for nautilus and one for master?

File: src/pybind/mgr/dashboard/frontend/package.json [hunk 1]
```diff
@@ -4,10 +4,11 @@
   "license": "MIT",
   "scripts": {
     "ng": "ng",
-    "start": "npm run env_build && ng serve --host 0.0.0.0",
-    "build": "npm run env_build && ng build",
+    "start": "npm run i18n-merge && npm run env_build && ng serve --host 0.0.0.0",
+    "build": "npm run i18n-merge && npm run env_build && ng build",
```
[the remark this answers is not in the document]
Comment (Exotelis) [reply src/pybind/mgr/dashboard/frontend/package.json]: I'm working on an implementation go get rid of the messages.xlf 
I just marked the pr as WIP until it has been resolved

**Comment by s0nea** [comment]
jenkins test docs

**Comment by LenzGr** [comment]
@Exotelis any news on this one? Do you still plan to update and merge this one?

**Comment by rjfd** [comment]
What's the current status? @tspmelo were your comments addressed?

I also have some questions.
Where does the ceph-dashboard-i18ntool code lives? I don't find it in the ceph repo. Why don't we add this tool code to the ceph repo, otherwise it will be hard to be maintained by other people.

**Comment by Exotelis** [comment]
> What's the current status? @tspmelo were your comments addressed?
> 
> I also have some questions.
> Where does the ceph-dashboard-i18ntool code lives? I don't find it in the ceph repo. Why don't we add this tool code to the ceph repo, otherwise it will be hard to be maintained by other people.

It is one of my private repositories. I wanted to have the tool available as npm package since everyone can use it who has an transifex project and wants to handle his translations files.
I've already considered to rename the tool but as long as it is not finished I'll just stick to 'ceph-dashboard-i18ntool'.

Furthermore, having an npm package makes the ceph dashboard independent and we could change the workflow of this process easier.

**Comment by smithfarm** [comment]
Are "having an npm package" and "living in the ceph repository" mutually exclusive? Having the dashboard depend on a tool that lives in someone's private repository does not sound like a good idea on the face of it.

**Comment by Exotelis** [comment]
> Are "having an npm package" and "living in the ceph repository" mutually exclusive? Having the dashboard depend on a tool that lives in someone's private repository does not sound like a good idea on the face of it.

No it is not. But the code in not related to ceph and can be used by anyone else who wants to work with xlf files and transifex. 
If someone wants to add new features or fork the project he would have to get engaged in the ceph community and this seems a little bit too excessive from my point of view (or at least he would have to get familiar with the contribution guidelines).

**Comment by Exotelis** [comment]
The tool can be found here: https://github.com/Exotelis/transifex-i18ntool

**Comment by Exotelis** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/frontend/package.json [hunk 2]
```diff
@@ -112,6 +117,7 @@
     "protractor": "5.4.2",
     "protractor-screenshoter-plugin": "0.10.3",
     "replace-in-file": "3.4.3",
+    "transifex-i18ntool": "^1.1.0",
```
Comment (tspmelo) [inline src/pybind/mgr/dashboard/frontend/package.json]:
```suggestion
    "transifex-i18ntool": "1.1.0",
```

Comment (Exotelis) [reply src/pybind/mgr/dashboard/frontend/package.json]: Good point, I'm going to change it

**Comment by tspmelo** [comment]
Please rebase.

**Pull request merged**